identity/identity.py

```diff
@@ -0,0 +1,117 @@
+import json
+import time
+
+import requests
+from jwcrypto import jwk, jwt
+
+from accounts.settings import accounts_settings
+
+
+JWKS_URL = f"{accounts_settings.PLATFORM_URL}/identity/jwks/"
+ATTEST_URL = f"{accounts_settings.PLATFORM_URL}/identity/attest/"
+REFRESH_URL = f"{accounts_settings.PLATFORM_URL}/identity/refresh/"
+
+
+class IdentityContainer:
+    refresh_jwt: jwt.JWT = None
+    access_jwt: jwt.JWT = None
+    platform_jwks: jwk.JWKSet = None
+
+
+container = IdentityContainer()
+
+
+def get_platform_jwks():
+    """
+    Download the JWKS from Platform to verify JWTS
+    """
+
+    try:
+        response = requests.get(JWKS_URL)
+        # For some reason this method wants a raw string instead of a python dictionary
+        container.platform_jwks = jwk.JWKSet.from_json(response.text)
+    except Exception:
+        container.platform_jwks = None
+
+
+def attest():
+    """
+    Perform the initial authentication (attest) with Platform using the Client ID
+    and Secret from DOT
+    """
+
+    response = requests.post(
+        ATTEST_URL, auth=(accounts_settings.CLIENT_ID, accounts_settings.CLIENT_SECRET)
+    )
+    content = response.json()
+    container.access_jwt = jwt.JWT(key=container.platform_jwks, jwt=content["access"])
+    container.refresh_jwt = jwt.JWT(key=container.platform_jwks, jwt=content["refresh"])
+
+
+def _refresh_if_outdated():
+    """
+    Refresh the access jwt if it is expired.
+    """
+    access_claims = json.loads(container.access_jwt.claims)
+    # only continue if our access jwt is expired
+    if time.time() < access_claims["exp"]:
+        return
+
+    auth_headers = {"Authorization": f"Bearer {container.refresh_jwt.serialize()}"}
+    response = requests.get(REFRESH_URL, headers=auth_headers)
```

Comment:
If the refresh token is invalid, then that means the original attest didn't work, right? In that case I think we should probably just error-out since it's unlikely a re-attest would succeed imo